js-exercises/better-string-lib/betterStringLib.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-plusplus */
+/* eslint-disable quotes */
+
+const regexSymbolWithCombiningMarks = /(\P{Mark})(\p{Mark}+)/gu;
+const regexSurrogatePair = /([\uD800-\uDBFF])([\uDC00-\uDFFF])/g;
+
+const reverse = (input) => {
+  const str = input
+    .replace(regexSymbolWithCombiningMarks, ($0, $1, $2) => reverse($2) + $1)
+    .replace(regexSurrogatePair, "$2$1");
+
+  const result = [];
+  let index = str.length;
+  while (index--) {
+    result.push(str.charAt(index));
+  }
+  return result.join("");
+};
```

Comment:
you can make use of inbuild functions to make your code clean and bugless.
you can check out : 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize